### DISCUSSION TITLE
Little confuse about msg's encoder

### DISCUSSION THREAD

**Discussion by RandalTeng** [wneessen/go-mail#162] [Q&A] [answered]
I had notice that, every header (like subject) set in msg, will be encoded by msg.encoder, I want to know why do this?
Is there any vulnerability about use origin content?
If there is a image, I can get why need encoder, it can't be set to subject though...

**Comment by RandalTeng** [comment]
There is a problem that if I have some 3 bit unicode string in subject, it will be convert to random code like this:
`msg.Subject(fmt.Sprintf("我是原始内容，日期：%s", cmd.timeMark.Format("01月02日")))`
It will become this:
![image](https://github.com/wneessen/go-mail/assets/13815562/b977a9fd-4c8c-4bd1-b026-9a0a503d333c)

Or there is something I have missed, tip me something, thanks.

**Reply by wneessen** [reply]
Yes, that is expected, since mail only supports US-ASCII we need to encode the UTF-8 string to an encoding that the mailserver and client can understand.

If your client does not display the expected output, your client might not be able to understand the encoding correctly.

I took the subject line you posted and send a test mail to myself and it was encoded correctly and is also displayed correctly:

This is the source code of the mail:
![image](https://github.com/wneessen/go-mail/assets/542696/6439eed7-0b74-49d8-986b-86090f75dc52)

which results in:
![image](https://github.com/wneessen/go-mail/assets/542696/94535075-6a33-49cd-b339-402c32a97676)

**Reply by RandalTeng** [reply]
> If your client does not display the expected output, your client might not be able to understand the encoding correctly.

So, basicly it is a mail client problem? There is no more changes need the server todo?

**Reply by wneessen** [reply]
Yes, most likely. If you have an alternative mail account with a modern web interface, like GMail, you could send a test mail to that account and verify if it works correctly. If not, please provide a full example and the provider in question, so I can try to reproduce.

**Comment by wneessen** [comment, marked as answer]
> I had notice that, every header (like subject) set in msg, will be encoded by msg.encoder, I want to know why do this? Is there any vulnerability about use origin content? If there is a image, I can get why need encoder, it can't be set to subject though...

That's basically how email works. Email only supports 7bit US-ASCII (with some exceptions in which mail servers support 8bit too). If you only use US-ASCII for your subjects and stuff, there is no need to encode it, but since go-mail supports full UTF-8, we need to encode the headers accordingly so that mail servers and clients can transfer and understand the content.

**Reply by RandalTeng** [reply]
Thanks for your answer, actually I'm not very understand how email server do with mail message.

**Reply by wneessen** [reply]
Pleasure, that's what go-mail is for, so you don't need to knowledge of mail servers but let it handle it for you ;)
